Chiazokam/iReporterApp
# feat (server): create user sign up endpoint

#### What does this PR do?
- creates user sign up endpoint
- creates middleware to test for whitespace in user input
- creates middleware to validate email input
- sets up json web token and use for user sign up
- seperates queries from controllers
- writes test for creating records

#### Description of Task to be completed?
- create user sign up endpoint
- create middleware to test for whitespace in user input
- create middleware to validate email input
- set up json web token and use for user sign up
- seperate queries from controllers
- write test for creating records

#### How should this be manually tested?
- Clone the repo - `https://github.com/Chiazokam/iReporterApp`
- Enter into the project directory - `cd iReporterApp`
- switch to this branch - `checkout ft-api-signup-database-162597347
- Start the server `npm run start:dev`
- Open `Postman`
- On `postman` Send a `POST` request to `http://localhost:3000/api/v1/users `
- You should receive a successful response

#### Any background context you want to provide?
- the POST endpoint payload should be structured like this- 
`
{
       "firstname": "Blame",
           "lastname": "Flower",
           "othername": "Mart",
           "email": "guru@gmail.com",
           "password": "father",
           "phone": "09035463739",
           "username": "Mame"
     }
`

#### What are the relevant pivotal tracker stories?
[#162597347](https://www.pivotaltracker.com/story/show/162597347)
